### PR TITLE
Enable Bucket Support in the CLI

### PR DESCRIPTION
**Pull request by jonathan-innis**
---

This checklist is used to make sure that common guidelines for a pull request are followed.

### General Guidelines

- [ ] Have you run `azdev style <YOUR_EXT>` locally? (`pip install azdev` required)
- [ ] Have you run `python scripts/ci/test_index.py -q` locally?

For new extensions:

- [ ] My extension description/summary conforms to the [Extension Summary Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/extensions/extension_summary_guidelines.md).

### About Extension Publish

There is a pipeline to automatically build, upload and publish extension wheels.  
Once your PR is merged into master branch, a new PR will be created to update `src/index.json` automatically.  
The precondition is to put your code inside this repo and upgrade the version in the PR but do not modify `src/index.json`.

### CODE REVIEW

File: src/k8s-configuration/azext_k8s_configuration/consts.py [hunk 1]
```diff
@@ -7,29 +7,53 @@
...
-GIT_REPOSITORY_REQUIRED_VALUES_MISSING_ERROR = "Error! Required property '{}' was not specified for kind 'GitRepository'"
-GIT_REPOSITORY_REQUIRED_VALUES_MISSING_HELP = "Add missing required property and try again"
+REPOSITORY_REF_TOO_MANY_VALUES_HELP = (
+    "Specifying more than one repository ref argument is invalid"
+)
+
+REQUIRED_VALUES_MISSING_ERROR = (
+    "Error! Required properties [{}] were not specified for kind '{}'"
+)
+REQUIRED_VALUES_MISSING_HELP = "Add the missing required properties and try again"
```
Comment (sophsoph321) [inline src/k8s-configuration/azext_k8s_configuration/consts.py:36]: nit: to be consistent, do you want to have all of the string messages be inside parentheses?

File: src/k8s-configuration/azext_k8s_configuration/utils.py [hunk 2]
```diff
@@ -55,29 +57,30 @@ def read_key_file(path, strip_newline=False):
...
 
 
 def parse_duration(duration):
     if not duration:
         return duration
-    regex = re.compile(r'((?P<hours>\d+?)h)?((?P<minutes>\d+?)m)?((?P<seconds>\d+?)s)?')
+    regex = re.compile(r"((?P<hours>\d+?)h)?((?P<minutes>\d+?)m)?((?P<seconds>\d+?)s)?")
     parts = regex.match(duration)
     parts = parts.groupdict()
+    print(parts)
```
Comment (sophsoph321) [inline src/k8s-configuration/azext_k8s_configuration/utils.py:83]: Was this for debugging purposes?

File: src/k8s-configuration/azext_k8s_configuration/consts.py (hunk 1 above)
Comment (jonathan-innis) [reply src/k8s-configuration/azext_k8s_configuration/consts.py:36]: It is a formatting thing. This is done by auto-formatter for strings that are too long

File: src/k8s-configuration/azext_k8s_configuration/utils.py (hunk 2 above)
Comment (jonathan-innis) [reply src/k8s-configuration/azext_k8s_configuration/utils.py:83]: Good catch, will remove

**Review by deeksha345 — APPROVED** [review]

**Pull request merged**